packages/ui-tree-browser/src/TreeBrowser/index.tsx

```diff
@@ -342,46 +316,46 @@ class TreeBrowser extends Component<TreeBrowserProps> {
     )
   }
 
-  // @ts-expect-error ts-migrate(7023) FIXME: 'getCollectionProps' implicitly has return type 'a... Remove this comment to see the full error message
-  getCollectionProps(collection) {
+  getCollectionProps(collection: Collection): CollectionProps {
     return {
       id: collection.id,
       name: collection.name,
-      descriptor: collection.descriptor,
-      expanded: this.getExpandedIndex(this.expanded, collection.id) >= 0,
-      items: this.getItems(collection),
       collections: this.getSubCollections(collection),
+      items: this.getItems(collection),
+      descriptor: collection.descriptor,
+      containerRef: collection.containerRef,
       renderBeforeItems: collection.renderBeforeItems,
       renderAfterItems: collection.renderAfterItems,
-      containerRef: collection.containerRef,
+      expanded: this.getExpandedIndex(this.expanded, collection.id) >= 0,
       isCollectionFlattened: this.getIsFlattened(collection)
     }
   }
 
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'expanded' implicitly has an 'any' type.
-  getExpandedIndex(expanded, id) {
-    // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'expanded' implicitly has an 'any' type.
+  getExpandedIndex(
+    expanded: (string | number | undefined)[],
+    id?: string | number
+  ) {
     return expanded.findIndex((expanded) => String(expanded) === String(id))
   }
 
   renderRoot() {
-    return this.collections.sort(this.props.sortOrder).map((collection, i) => (
-      <TreeCollection
-        key={i}
-        // @ts-expect-error ts-migrate(2554) FIXME: Expected 3 arguments, but got 2.
-        {...pickProps(omitProps(this.props), TreeCollection.allowedProps)}
-        {...this.getCollectionProps(collection)}
-        // @ts-expect-error ts-migrate(2339) FIXME: Property 'selection' does not exist on type 'Reado... Remove this comment to see the full error message
-        selection={this.state.selection}
-        onItemClick={this.handleItemClick}
-        onCollectionClick={this.handleCollectionClick}
-        onKeyDown={this.handleKeyDown}
-        numChildren={this.collections.length}
-        level={1}
-        position={1}
-        renderContent={this.props.renderContent}
-      />
-    ))
+    return this.collections
+      .sort(this.props.sortOrder)
+      .map((collection, i) => (
+        <TreeCollection
+          key={i}
+          {...pickProps(this.props, TreeCollection.allowedProps)}
```

Comment:
I think the `omitProps` here was supposed to cut all the automatically excluded props (e.g. withStyle props, className, etc). We could make the second parameter optional :) It might be good to keep it, because if in the future we add "className" as an allowedProp on TreeCollection, then TreeBrowser will pass it's own themeOverrides to TreeCollection, etc

Reply:
yep, you are right. I've made the second param of `omitProps` optional